src/base_circuitpython/displayio/group.py

```diff
@@ -32,19 +39,50 @@ def append(self, item):
             raise ValueError(CONSTANTS.LAYER_ALREADY_IN_GROUP)
 
         self.__contents.append(item)
-        item.in_group = True
-        if self.auto_write:
-            self.draw(show=True)
+        item.parent = self
+
+        self.__elem_changed()
+
+    def __elem_changed(self):
+        # ensure that this group is what the board is currently showing
+        # otherwise, don't bother to draw it
+        if (
+            self.__auto_write
+            and self.__check_active_group_ref
+            and board.DISPLAY.active_group == self
+        ):
+            self.draw()
+
+        elif self.in_group:
+
+            # if a sub-group is modified,
+            # propagate to top level to
+            # see if one of the parents are the
+            # current active group
+            self.parent.__elem_changed()
 
     def __getitem__(self, index):
         return self.__contents[index]
 
     def __setitem__(self, index, val):
+        old_val = self.__contents[index]
+
         self.__contents[index] = val
 
-    def draw(self, x=0, y=0, scale=None, show=False):
+        if old_val != val:
+            self.__elem_changed()
+
+    def draw(self, img=None, x=0, y=0, scale=None, show=True):
+

```

Comment:
_&ast;nit&ast;_
I see some functions have a new line after the signature and some don't. Perhaps be more consistent?